hack/verify-shellcheck.sh

```diff
@@ -41,6 +47,22 @@ join_by() {
 SHELLCHECK_DISABLED="$(join_by , "${disabled[@]}")"
 readonly SHELLCHECK_DISABLED
 
+# creates the shellcheck container for later use
+create_container () {
+  # TODO(bentheelder): this is a performance hack, we create the container with
+  # a sleep MAX_INT32 so that it is effectively paused.
+  # We then repeatedly exec to it to run each shellcheck, and later rm it when
+  # we're done.
+  # This is incredibly much faster than creating a container for each shellcheck
+  # call ...
+  docker run --name "${SHELLCHECK_CONTAINER}" -d --rm -v "${KUBE_ROOT}:${KUBE_ROOT}" -w "${KUBE_ROOT}" --entrypoint="sleep" "${SHELLCHECK_IMAGE}" 2147483647
+}
+# removes the shellcheck container
+remove_container () {
```

Comment:
do we want to call this from a trap?

Reply:
we do below (?)

Reply:
https://github.com/kubernetes/kubernetes/pull/72862/files#diff-21849b598a2db215f292c1bd4620e454R131 calls this in a trap, we only setup the trap if we're in container mode rather than host binary mode, but we set the trap before creating the container.

Reply:
cool thanks